src/js/providers/utils/quality-labels.ts

```diff
@@ -1,24 +1,36 @@
+import type { ProviderLevel } from 'providers/data-normalizer';
+import type { GenericObject } from 'types/generic.type';
+
+type QualityLabels = {
+    [key: number]: string;
+};
+
 // Try and find a corresponding custom label. If there are no custom labels, create one using height, bandwidth, or both
-export function generateLabel(level, qualityLabels, redundant) {
+export function generateLabel(level: ProviderLevel | undefined, qualityLabels: QualityLabels | null, redundant?: boolean): string {
```

Comment:
level?: ProviderLevel

Reply:
nvm. It's to avoid re-ordering right?

Reply:
Actually after taking a look, it looks like that can be since it could be `qualityLabels?`. It looks like the qualityLabels passed in come from the player config, so they could be undefined.
Nice catch!